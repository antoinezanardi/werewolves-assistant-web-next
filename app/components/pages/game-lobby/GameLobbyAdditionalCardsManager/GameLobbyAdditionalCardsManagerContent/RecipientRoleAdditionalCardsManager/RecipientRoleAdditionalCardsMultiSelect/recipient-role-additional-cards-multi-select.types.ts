import type { CreateGameAdditionalCardDto } from "~/composables/api/game/dto/create-game/create-game-additional-card/create-game-additional-card.dto";
import type { GameAdditionalCardRecipientRoleName } from "~/composables/api/game/types/game-additional-card/game-additional-card.types";

type RecipientRoleAdditionalCardsMultiSelectProps = {
  recipientRoleName: GameAdditionalCardRecipientRoleName;
};

type LabeledCreateGameAdditionalCardDto = CreateGameAdditionalCardDto & { label: string };

export type {
  RecipientRoleAdditionalCardsMultiSelectProps,
  LabeledCreateGameAdditionalCardDto,
};