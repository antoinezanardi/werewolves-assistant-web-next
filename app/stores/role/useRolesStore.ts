import type { AsyncDataRequestStatus } from "nuxt/app";
import { defineStore } from "pinia";
import { ref } from "vue";
import type { GameAdditionalCardRecipientRoleName } from "~/composables/api/game/types/game-additional-card/game-additional-card.types";

import type { Role } from "~/composables/api/role/types/role.class";
import type { RoleName, RoleSide } from "~/composables/api/role/types/role.types";
import { useFetchRoles } from "~/composables/api/role/useFetchRoles";
import { StoreIds } from "~/stores/enums/store.enum";

const useRolesStore = defineStore(StoreIds.ROLES, () => {
  const { fetchRoles } = useFetchRoles();

  const fetchingRoleStatus = ref<AsyncDataRequestStatus>("idle");
  const roles = ref<Role[] | null>(null);

  async function fetchAndSetRoles(): Promise<void> {
    fetchingRoleStatus.value = "pending";
    roles.value = await fetchRoles();
    fetchingRoleStatus.value = "success";
  }

  function getRoleWithNameInRoles(roleName: RoleName): Role | undefined {
    return roles.value?.find(role => role.name === roleName);
  }

  function getRolesForRecipientRoleName(recipientRoleName: GameAdditionalCardRecipientRoleName): Role[] {
    return roles.value?.filter(role => role.additionalCardsEligibleRecipients?.includes(recipientRoleName)) ?? [];
  }

  function getRoleSideForRoleName(roleName: RoleName): RoleSide | undefined {
    return getRoleWithNameInRoles(roleName)?.side;
  }
  return {
    roles,
    fetchingRoleStatus,
    fetchAndSetRoles,
    getRoleWithNameInRoles,
    getRolesForRecipientRoleName,
    getRoleSideForRoleName,
  };
});

export { useRolesStore };