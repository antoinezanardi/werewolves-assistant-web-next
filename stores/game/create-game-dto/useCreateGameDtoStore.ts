import { defineStore } from "pinia";
import { ref } from "vue";

import { CreateGamePlayerDto } from "~/composables/api/game/dto/create-game/create-game-player/create-game-player.dto";
import { CreateGameDto } from "~/composables/api/game/dto/create-game/create-game.dto";
import { StoreIds } from "~/stores/enums/store.enum";

const useCreateGameDtoStore = defineStore(StoreIds.CREATE_GAME_DTO, () => {
  const createGameDto = ref<CreateGameDto>(CreateGameDto.create({ players: [] }));

  function setCreateGameDto(createGameDtoValue: CreateGameDto): void {
    createGameDto.value = CreateGameDto.create(createGameDtoValue);
  }

  function resetCreateGameDto(): void {
    createGameDto.value = CreateGameDto.create({ players: [] });
  }

  function addPlayerToCreateGameDto(player: CreateGamePlayerDto): void {
    createGameDto.value.players.push(CreateGamePlayerDto.create(player));
  }

  function updatePlayerInCreateGameDto(player: CreateGamePlayerDto): void {
    const playerIndex = createGameDto.value.players.findIndex(({ name }) => name === player.name);
    if (playerIndex !== -1) {
      createGameDto.value.players.splice(playerIndex, 1, CreateGamePlayerDto.create(player));
    }
  }

  function setPlayersToCreateGameDto(players: CreateGamePlayerDto[]): void {
    createGameDto.value.players = players.map(player => CreateGamePlayerDto.create(player));
  }

  function removePlayerFromCreateGameDto(playerName: string): void {
    const playerIndex = createGameDto.value.players.findIndex(player => player.name === playerName);
    if (playerIndex !== -1) {
      createGameDto.value.players = createGameDto.value.players.toSpliced(playerIndex, 1);
    }
  }
  return {
    createGameDto,
    setCreateGameDto,
    resetCreateGameDto,
    addPlayerToCreateGameDto,
    updatePlayerInCreateGameDto,
    setPlayersToCreateGameDto,
    removePlayerFromCreateGameDto,
  };
});

export { useCreateGameDtoStore };