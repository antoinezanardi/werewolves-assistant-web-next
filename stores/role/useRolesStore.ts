import type { AsyncDataRequestStatus } from "#app/composables/asyncData";
import { defineStore } from "pinia";
import { ref } from "vue";

import type { Role } from "~/composables/api/role/types/role.class";
import type { RoleName } from "~/composables/api/role/types/role.types";
import { useFetchRoles } from "~/composables/api/role/useFetchRoles";
import { StoreIds } from "~/stores/enums/store.enum";

const useRolesStore = defineStore(StoreIds.ROLES, () => {
  const { fetchRoles } = useFetchRoles();

  const fetchingRoleStatus = ref<AsyncDataRequestStatus>("idle");
  const roles = ref<Role[] | null>(null);

  async function fetchAndSetRoles(): Promise<void> {
    fetchingRoleStatus.value = "pending";
    roles.value = await fetchRoles();
    fetchingRoleStatus.value = "success";
  }

  function getRoleWithNameInRoles(roleName: RoleName): Role | undefined {
    return roles.value?.find(role => role.name === roleName);
  }
  return {
    roles,
    fetchingRoleStatus,
    fetchAndSetRoles,
    getRoleWithNameInRoles,
  };
});

export { useRolesStore };