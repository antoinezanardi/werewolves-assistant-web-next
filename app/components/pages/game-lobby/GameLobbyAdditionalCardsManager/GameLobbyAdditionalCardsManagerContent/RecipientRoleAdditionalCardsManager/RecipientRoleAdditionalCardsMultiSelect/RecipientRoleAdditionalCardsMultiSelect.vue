<template>
  <div id="recipient-role-additional-cards-multi-select-container">
    <PrimeVueMultiSelect
      id="recipient-role-additional-cards-multi-select"
      v-model="selectedAdditionalCards"
      auto-filter-focus
      class="w-full"
      display="chip"
      :empty-filter-message="$t('components.RecipientRoleAdditionalCardsMultiSelect.noRoleFound')"
      filter
      :filter-placeholder="$t('components.RecipientRoleAdditionalCardsMultiSelect.searchRole')"
      highlight-on-select
      :invalid="!isAtLeastOneAdditionalCardSelected"
      option-label="label"
      :options="availableAdditionalCards"
      :placeholder="$t('components.RecipientRoleAdditionalCardsMultiSelect.pickOneToFiveCards')"
      :pt="{ 'labelContainer': 'flex justify-center items-center h-20' }"
      reset-filter-on-hide
      :selection-limit="5"
      :show-toggle-all="false"
    >
      <template #option="{ option }">
        <div class="align-options-center flex">
          <RoleImage
            class="me-2"
            :role-name="option.roleName"
            size="small"
            sizes="30"
          />

          <div>{{ option.label }}</div>
        </div>
      </template>

      <template #chip="{ value }">
        <RoleImage
          class="me-2"
          :role-name="value.roleName"
          size="small"
        />

        <span>{{ value.label }}</span>
      </template>
    </PrimeVueMultiSelect>
  </div>
</template>

<script setup lang="ts">
import { storeToRefs } from "pinia";
import type { LabeledCreateGameAdditionalCardDto, RecipientRoleAdditionalCardsMultiSelectProps } from "~/components/pages/game-lobby/GameLobbyAdditionalCardsManager/GameLobbyAdditionalCardsManagerContent/RecipientRoleAdditionalCardsManager/RecipientRoleAdditionalCardsMultiSelect/recipient-role-additional-cards-multi-select.types";
import RoleImage from "~/components/shared/role/RoleImage/RoleImage.vue";
import { CreateGameAdditionalCardDto } from "~/composables/api/game/dto/create-game/create-game-additional-card/create-game-additional-card.dto";
import type { RoleName } from "~/composables/api/role/types/role.types";
import { useRoleName } from "~/composables/api/role/useRoleName";
import { useCreateGameDtoStore } from "~/stores/game/create-game-dto/useCreateGameDtoStore";
import { useRolesStore } from "~/stores/role/useRolesStore";

const props = defineProps<RecipientRoleAdditionalCardsMultiSelectProps>();

const createGameDtoStore = useCreateGameDtoStore();
const { getAdditionalCardsForRecipientInCreateGameDto } = createGameDtoStore;
const { createGameDto } = storeToRefs(createGameDtoStore);

const rolesStore = useRolesStore();
const { getRolesForRecipientRoleName } = rolesStore;
const { getRoleNameLabel } = useRoleName();

const selectedAdditionalCards = computed<CreateGameAdditionalCardDto[]>({
  get: () => getAdditionalCardsForRecipientInCreateGameDto(props.recipientRoleName),
  set: (value: CreateGameAdditionalCardDto[]) => {
    createGameDtoStore.setAdditionalCardsForRecipientInCreateGameDto(value, props.recipientRoleName);
  },
});

const isAtLeastOneAdditionalCardSelected = computed<boolean>(() => selectedAdditionalCards.value.length > 0);

const availableAdditionalCards = computed<LabeledCreateGameAdditionalCardDto[]>(() => {
  const roles = getRolesForRecipientRoleName(props.recipientRoleName);
  const availableRoles = roles.filter(({ name }) => isRoleAvailableInCreateGameDto(name));

  const availableCards = availableRoles.map(({ name }) => ({
    ...CreateGameAdditionalCardDto.create({
      recipient: props.recipientRoleName,
      roleName: name,
    }),
    label: getRoleNameLabel(name),
  }));

  return availableCards.toSorted((cardA, cardB) => cardA.label.localeCompare(cardB.label));
});

function isRoleAvailableInCreateGameDto(roleName: RoleName): boolean {
  const isRoleTakenAmongPlayers = createGameDto.value.players.some(player => player.role.name === roleName);
  const isRoleTakenAmongAdditionalCards = createGameDto.value.additionalCards?.some(additionalCard => additionalCard.roleName === roleName &&
    additionalCard.recipient !== props.recipientRoleName) === true;

  return !isRoleTakenAmongPlayers && !isRoleTakenAmongAdditionalCards;
}
</script>