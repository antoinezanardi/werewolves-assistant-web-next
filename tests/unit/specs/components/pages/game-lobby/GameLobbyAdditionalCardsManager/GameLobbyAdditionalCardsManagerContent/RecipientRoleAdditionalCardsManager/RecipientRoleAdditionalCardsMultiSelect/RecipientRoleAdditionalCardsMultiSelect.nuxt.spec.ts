import { createTestingPinia } from "@pinia/testing";
import { createFakeCreateGameAdditionalCardDto } from "@tests/unit/utils/factories/composables/api/game/dto/create-game/create-game-additional-card/create-game-additional-card.dto.factory";
import { createFakeCreateGamePlayerRoleDto } from "@tests/unit/utils/factories/composables/api/game/dto/create-game/create-game-player/create-game-player-role/create-game-player-role.dto.factory";
import { createFakeCreateGamePlayerDto } from "@tests/unit/utils/factories/composables/api/game/dto/create-game/create-game-player/create-game-player.dto.factory";
import { createFakeCreateGameDto } from "@tests/unit/utils/factories/composables/api/game/dto/create-game/create-game.dto.factory";
import { createFakeRole } from "@tests/unit/utils/factories/composables/api/role/role.factory";
import type { mount } from "@vue/test-utils";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";
import type VueUseCore from "@vueuse/core";
import type Button from "primevue/button";
import type MultiSelect from "primevue/multiselect";
import type { MultiSelectProps } from "primevue/multiselect";
import type { LabeledCreateGameAdditionalCardDto, RecipientRoleAdditionalCardsMultiSelectProps } from "~/components/pages/game-lobby/GameLobbyAdditionalCardsManager/GameLobbyAdditionalCardsManagerContent/RecipientRoleAdditionalCardsManager/RecipientRoleAdditionalCardsMultiSelect/recipient-role-additional-cards-multi-select.types";
import RecipientRoleAdditionalCardsMultiSelect from "~/components/pages/game-lobby/GameLobbyAdditionalCardsManager/GameLobbyAdditionalCardsManagerContent/RecipientRoleAdditionalCardsManager/RecipientRoleAdditionalCardsMultiSelect/RecipientRoleAdditionalCardsMultiSelect.vue";
import type { CreateGameAdditionalCardDto } from "~/composables/api/game/dto/create-game/create-game-additional-card/create-game-additional-card.dto";
import { StoreIds } from "~/stores/enums/store.enum";
import { useCreateGameDtoStore } from "~/stores/game/create-game-dto/useCreateGameDtoStore";
import { useRolesStore } from "~/stores/role/useRolesStore";

const hoistedMocks = vi.hoisted(() => ({
  useBreakpoints: {
    smaller: vi.fn(),
  },
}));

vi.mock("@vueuse/core", async importOriginal => ({
  ...await importOriginal<typeof VueUseCore>(),
  useBreakpoints: (): typeof hoistedMocks.useBreakpoints => hoistedMocks.useBreakpoints,
}));

describe("Recipient Role Additional Cards Multi Select Component", () => {
  const defaultRoles = [
    createFakeRole({
      name: "seer",
      additionalCardsEligibleRecipients: ["thief"],
    }),
    createFakeRole({
      name: "hunter",
      additionalCardsEligibleRecipients: ["actor"],
    }),
    createFakeRole({
      name: "defender",
      additionalCardsEligibleRecipients: ["thief", "actor"],
    }),
    createFakeRole({
      name: "witch",
    }),
    createFakeRole({
      name: "idiot",
      additionalCardsEligibleRecipients: ["thief"],
    }),
    createFakeRole({
      name: "pied-piper",
      additionalCardsEligibleRecipients: ["thief"],
    }),
    createFakeRole({
      name: "accursed-wolf-father",
      additionalCardsEligibleRecipients: ["thief"],
    }),
  ];
  const defaultCreateFakeCreateGameDto = createFakeCreateGameDto({
    players: [
      createFakeCreateGamePlayerDto({ role: createFakeCreateGamePlayerRoleDto({ name: "idiot" }) }),
      createFakeCreateGamePlayerDto({ role: createFakeCreateGamePlayerRoleDto({ name: "villager" }) }),
    ],
    additionalCards: [
      createFakeCreateGameAdditionalCardDto({
        roleName: "defender",
        recipient: "actor",
      }),
      createFakeCreateGameAdditionalCardDto({
        roleName: "pied-piper",
        recipient: "thief",
      }),
    ],
  });
  const testingPinia = {
    initialState: {
      [StoreIds.CREATE_GAME_DTO]: {
        createGameDto: defaultCreateFakeCreateGameDto,
      },
      [StoreIds.ROLES]: {
        roles: defaultRoles,
      },
    },
    stubActions: false,
  };
  const defaultProps: RecipientRoleAdditionalCardsMultiSelectProps = {
    recipientRoleName: "thief",
  };
  let wrapper: ReturnType<typeof mount<typeof RecipientRoleAdditionalCardsMultiSelect>>;

  async function mountRecipientRoleAdditionalCardsMultiSelectComponent(options: ComponentMountingOptions<typeof RecipientRoleAdditionalCardsMultiSelect> = {}):
  Promise<ReturnType<typeof mount<typeof RecipientRoleAdditionalCardsMultiSelect>>> {
    return mountSuspendedComponent(RecipientRoleAdditionalCardsMultiSelect, {
      global: {
        plugins: [createTestingPinia(testingPinia)],
        stubs: {
          MultiSelect: false,
        },
      },
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    hoistedMocks.useBreakpoints.smaller.mockReturnValue(ref(false));
    wrapper = await mountRecipientRoleAdditionalCardsMultiSelectComponent();
    const createGameDtoStore = useCreateGameDtoStore();
    createGameDtoStore.createGameDto = createFakeCreateGameDto(defaultCreateFakeCreateGameDto);
    const rolesStore = useRolesStore();
    rolesStore.roles = defaultRoles;
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should match snapshot when rendered without shallow rendering.", async() => {
    wrapper = await mountRecipientRoleAdditionalCardsMultiSelectComponent({ shallow: false });

    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should match snapshot when rendered without shallow rendering and screen is smaller than md.", async() => {
    hoistedMocks.useBreakpoints.smaller.mockReturnValue(ref(true));
    wrapper = await mountRecipientRoleAdditionalCardsMultiSelectComponent({ shallow: false });

    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Multi Select", () => {
    it("should pass recipient available additional cards to multi select when role is not taken by players nor by existing additional cards recipients.", () => {
      const expectedRecipientAdditionalCards: LabeledCreateGameAdditionalCardDto[] = [
        {
          ...createFakeCreateGameAdditionalCardDto({
            roleName: "accursed-wolf-father",
            recipient: "thief",
          }),
          label: "shared.role.name.accursed-wolf-father",
        },
        {
          ...createFakeCreateGameAdditionalCardDto({
            roleName: "pied-piper",
            recipient: "thief",
          }),
          label: "shared.role.name.pied-piper",
        },
        {
          ...createFakeCreateGameAdditionalCardDto({
            roleName: "seer",
            recipient: "thief",
          }),
          label: "shared.role.name.seer",
        },
      ];
      const multiSelect = wrapper.findComponent<typeof MultiSelect>("#recipient-role-additional-cards-multi-select");
      const props = multiSelect.props() as MultiSelectProps;

      expect(props.options).toStrictEqual<LabeledCreateGameAdditionalCardDto[]>(expectedRecipientAdditionalCards);
    });

    it("should pass more recipient available additional cards to multi select when there are no additional cards.", async() => {
      const createGameDtoStore = useCreateGameDtoStore();
      createGameDtoStore.createGameDto.additionalCards = undefined;
      await nextTick();
      const expectedRecipientAdditionalCards: LabeledCreateGameAdditionalCardDto[] = [
        {
          ...createFakeCreateGameAdditionalCardDto({
            roleName: "accursed-wolf-father",
            recipient: "thief",
          }),
          label: "shared.role.name.accursed-wolf-father",
        },
        {
          ...createFakeCreateGameAdditionalCardDto({
            roleName: "defender",
            recipient: "thief",
          }),
          label: "shared.role.name.defender",
        },
        {
          ...createFakeCreateGameAdditionalCardDto({
            roleName: "pied-piper",
            recipient: "thief",
          }),
          label: "shared.role.name.pied-piper",
        },
        {
          ...createFakeCreateGameAdditionalCardDto({
            roleName: "seer",
            recipient: "thief",
          }),
          label: "shared.role.name.seer",
        },
      ];
      const multiSelect = wrapper.findComponent<typeof MultiSelect>("#recipient-role-additional-cards-multi-select");
      const props = multiSelect.props() as MultiSelectProps;

      expect(props.options).toStrictEqual<LabeledCreateGameAdditionalCardDto[]>(expectedRecipientAdditionalCards);
    });

    it("should translate empty filter message when rendered.", () => {
      const multiSelect = wrapper.findComponent<typeof MultiSelect>("#recipient-role-additional-cards-multi-select");
      const props = multiSelect.props() as MultiSelectProps;

      expect(props.emptyFilterMessage).toBe("No role found");
    });

    it("should translate filter placeholder when rendered.", () => {
      const multiSelect = wrapper.findComponent<typeof MultiSelect>("#recipient-role-additional-cards-multi-select");
      const props = multiSelect.props() as MultiSelectProps;

      expect(props.filterPlaceholder).toBe("Search for a role");
    });

    it("should pass selected recipient additional cards to multi select when recipient has selected cards already.", () => {
      const expectedRecipientAdditionalCards = [
        createFakeCreateGameAdditionalCardDto({
          roleName: "pied-piper",
          recipient: "thief",
        }),
      ];
      const multiSelect = wrapper.findComponent<typeof MultiSelect>("#recipient-role-additional-cards-multi-select");
      const props = multiSelect.props() as MultiSelectProps;

      expect(props.modelValue).toStrictEqual<CreateGameAdditionalCardDto[]>(expectedRecipientAdditionalCards);
    });

    it("should update create game dto additional cards when multi select emit update model value event.", async() => {
      const multiSelect = wrapper.findComponent<typeof MultiSelect>("#recipient-role-additional-cards-multi-select");
      const expectedRecipientAdditionalCards = [
        createFakeCreateGameAdditionalCardDto({
          roleName: "seer",
          recipient: "thief",
        }),
      ];
      const createGameDtoStore = useCreateGameDtoStore();
      multiSelect.vm.$emit("update:modelValue", expectedRecipientAdditionalCards);
      await nextTick();

      expect(createGameDtoStore.setAdditionalCardsForRecipientInCreateGameDto).toHaveBeenCalledExactlyOnceWith(expectedRecipientAdditionalCards, "thief");
    });

    it("should set invalid when there are no additional cards for recipient role.", async() => {
      const createGameDtoStore = useCreateGameDtoStore();
      createGameDtoStore.createGameDto.additionalCards = [];
      await nextTick();
      const multiSelect = wrapper.findComponent<typeof MultiSelect>("#recipient-role-additional-cards-multi-select");
      const props = multiSelect.props() as MultiSelectProps;

      expect(props.invalid).toBeTruthy();
    });

    it("should not set invalid when there are additional cards for recipient role.", () => {
      const multiSelect = wrapper.findComponent<typeof MultiSelect>("#recipient-role-additional-cards-multi-select");
      const props = multiSelect.props() as MultiSelectProps;

      expect(props.invalid).toBeFalsy();
    });

    it("should remove card when click on remove button.", async() => {
      const createGameDtoStore = useCreateGameDtoStore();
      createGameDtoStore.createGameDto.additionalCards = [
        createFakeCreateGameAdditionalCardDto({
          roleName: "pied-piper",
          recipient: "thief",
        }),
        createFakeCreateGameAdditionalCardDto({
          roleName: "seer",
          recipient: "thief",
        }),
      ];
      await nextTick();
      const removeAdditionalCardButtons = wrapper.findAllComponents<typeof Button>(".remove-additional-card-button");
      const secondButton = removeAdditionalCardButtons[1];
      await secondButton.trigger("click");
      const newAdditionalCards = [
        createFakeCreateGameAdditionalCardDto({
          roleName: "pied-piper",
          recipient: "thief",
        }),
      ];

      expect(createGameDtoStore.setAdditionalCardsForRecipientInCreateGameDto).toHaveBeenCalledExactlyOnceWith(newAdditionalCards, "thief");
    });
  });
});