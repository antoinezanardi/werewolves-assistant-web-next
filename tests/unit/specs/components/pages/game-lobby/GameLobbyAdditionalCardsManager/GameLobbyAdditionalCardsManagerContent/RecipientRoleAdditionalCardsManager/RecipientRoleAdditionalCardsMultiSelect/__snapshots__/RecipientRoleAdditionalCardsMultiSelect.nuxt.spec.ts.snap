// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Recipient Role Additional Cards Multi Select Component > should match snapshot when rendered without shallow rendering and screen is smaller than md. 1`] = `
"<div id="recipient-role-additional-cards-multi-select-container">
  <div class="p-multiselect p-component p-inputwrapper p-multiselect-display-chip p-inputwrapper-filled w-full" id="recipient-role-additional-cards-multi-select" data-testid="recipient-role-additional-cards-multi-select-thief" data-pc-name="multiselect" data-pc-section="root" pc7="">
    <div class="p-hidden-accessible" data-pc-section="hiddeninputcontainer" data-p-hidden-accessible="true"><input type="text" readonly="" placeholder="Pick one to five cards" tabindex="0" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-controls="recipient-role-additional-cards-multi-select_list" data-pc-section="hiddeninput"></div>
    <div class="p-multiselect-label-container flex justify-center items-center md:h-20" data-pc-section="labelcontainer">
      <div class="p-multiselect-label" data-pc-section="label"><span class="p-multiselect-chip-item flex items-center bg-gray-800 rounded md:gap-2 md:px-2 md:py-1" data-pc-section="chipitem"><img width="35" height="35" alt="shared.role.name.pied-piper" data-nuxt-img="" class="border-4 border-gray-800 role-image rounded-lg h-[35pxpx] w-[35pxpx]" src="/svg/misc/infinite-spinner.svg"><span class="hidden lg:inline"></span><button class="p-button p-component p-button-icon-only !hidden md:!inline p-button-rounded p-button-sm p-button-text remove-additional-card-button" type="button" data-pc-name="button" data-p-disabled="false" data-testid="recipient-role-additional-cards-multi-select-remove-pied-piper-for-thief" data-pc-section="root" data-pd-ripple="true" style="overflow: hidden; position: relative;" pc6=""><svg class="svg-inline--fa fa-times" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="times" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512">
            <path class="" fill="currentColor" d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"></path>
          </svg><span class="p-button-label" data-pc-section="label"> </span>
          <!----><span role="presentation" aria-hidden="true" data-p-ink="true" data-p-ink-active="false" class="p-ink" pd2="" data-pc-name="ripple" data-pc-section="root"></span>
        </button></span>
        <!---->
      </div>
    </div>
    <div class="p-multiselect-dropdown" data-pc-section="dropdown"><svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg" class="p-icon p-multiselect-dropdown-icon" aria-hidden="true" data-pc-section="dropdownicon">
        <path d="M7.01744 10.398C6.91269 10.3985 6.8089 10.378 6.71215 10.3379C6.61541 10.2977 6.52766 10.2386 6.45405 10.1641L1.13907 4.84913C1.03306 4.69404 0.985221 4.5065 1.00399 4.31958C1.02276 4.13266 1.10693 3.95838 1.24166 3.82747C1.37639 3.69655 1.55301 3.61742 1.74039 3.60402C1.92777 3.59062 2.11386 3.64382 2.26584 3.75424L7.01744 8.47394L11.769 3.75424C11.9189 3.65709 12.097 3.61306 12.2748 3.62921C12.4527 3.64535 12.6199 3.72073 12.7498 3.84328C12.8797 3.96582 12.9647 4.12842 12.9912 4.30502C13.0177 4.48162 12.9841 4.662 12.8958 4.81724L7.58083 10.1322C7.50996 10.2125 7.42344 10.2775 7.32656 10.3232C7.22968 10.3689 7.12449 10.3944 7.01744 10.398Z" fill="currentColor"></path>
      </svg></div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
</div>"
`;

exports[`Recipient Role Additional Cards Multi Select Component > should match snapshot when rendered without shallow rendering. 1`] = `
"<div id="recipient-role-additional-cards-multi-select-container">
  <div class="p-multiselect p-component p-inputwrapper p-multiselect-display-chip p-inputwrapper-filled w-full" id="recipient-role-additional-cards-multi-select" data-testid="recipient-role-additional-cards-multi-select-thief" data-pc-name="multiselect" data-pc-section="root" pc4="">
    <div class="p-hidden-accessible" data-pc-section="hiddeninputcontainer" data-p-hidden-accessible="true"><input type="text" readonly="" placeholder="Pick one to five cards" tabindex="0" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-controls="recipient-role-additional-cards-multi-select_list" data-pc-section="hiddeninput"></div>
    <div class="p-multiselect-label-container flex justify-center items-center md:h-20" data-pc-section="labelcontainer">
      <div class="p-multiselect-label" data-pc-section="label"><span class="p-multiselect-chip-item flex items-center bg-gray-800 rounded md:gap-2 md:px-2 md:py-1" data-pc-section="chipitem"><img width="50" height="50" alt="shared.role.name.pied-piper" data-nuxt-img="" class="border-4 border-gray-800 role-image rounded-lg h-[50pxpx] w-[50pxpx]" src="/svg/misc/infinite-spinner.svg"><span class="hidden lg:inline"></span><button class="p-button p-component p-button-icon-only !hidden md:!inline p-button-rounded p-button-sm p-button-text remove-additional-card-button" type="button" data-pc-name="button" data-p-disabled="false" data-testid="recipient-role-additional-cards-multi-select-remove-pied-piper-for-thief" data-pc-section="root" data-pd-ripple="true" style="overflow: hidden; position: relative;" pc3=""><svg class="svg-inline--fa fa-times" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="times" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512">
            <path class="" fill="currentColor" d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"></path>
          </svg><span class="p-button-label" data-pc-section="label"> </span>
          <!----><span role="presentation" aria-hidden="true" data-p-ink="true" data-p-ink-active="false" class="p-ink" pd1="" data-pc-name="ripple" data-pc-section="root"></span>
        </button></span>
        <!---->
      </div>
    </div>
    <div class="p-multiselect-dropdown" data-pc-section="dropdown"><svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg" class="p-icon p-multiselect-dropdown-icon" aria-hidden="true" data-pc-section="dropdownicon">
        <path d="M7.01744 10.398C6.91269 10.3985 6.8089 10.378 6.71215 10.3379C6.61541 10.2977 6.52766 10.2386 6.45405 10.1641L1.13907 4.84913C1.03306 4.69404 0.985221 4.5065 1.00399 4.31958C1.02276 4.13266 1.10693 3.95838 1.24166 3.82747C1.37639 3.69655 1.55301 3.61742 1.74039 3.60402C1.92777 3.59062 2.11386 3.64382 2.26584 3.75424L7.01744 8.47394L11.769 3.75424C11.9189 3.65709 12.097 3.61306 12.2748 3.62921C12.4527 3.64535 12.6199 3.72073 12.7498 3.84328C12.8797 3.96582 12.9647 4.12842 12.9912 4.30502C13.0177 4.48162 12.9841 4.662 12.8958 4.81724L7.58083 10.1322C7.50996 10.2125 7.42344 10.2775 7.32656 10.3232C7.22968 10.3689 7.12449 10.3944 7.01744 10.398Z" fill="currentColor"></path>
      </svg></div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
</div>"
`;

exports[`Recipient Role Additional Cards Multi Select Component > should match snapshot when rendered. 1`] = `
"<div id="recipient-role-additional-cards-multi-select-container">
  <div class="p-multiselect p-component p-inputwrapper p-multiselect-display-chip p-inputwrapper-filled w-full" id="recipient-role-additional-cards-multi-select" data-testid="recipient-role-additional-cards-multi-select-thief" data-pc-name="multiselect" data-pc-section="root" pc1="">
    <div class="p-hidden-accessible" data-pc-section="hiddeninputcontainer" data-p-hidden-accessible="true"><input type="text" readonly="" placeholder="Pick one to five cards" tabindex="0" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-controls="recipient-role-additional-cards-multi-select_list" data-pc-section="hiddeninput"></div>
    <div class="p-multiselect-label-container flex justify-center items-center md:h-20" data-pc-section="labelcontainer">
      <div class="p-multiselect-label" data-pc-section="label"><span class="p-multiselect-chip-item flex items-center bg-gray-800 rounded md:gap-2 md:px-2 md:py-1" data-pc-section="chipitem"><role-image-stub rolename="pied-piper" sizes="50px" definition="small"></role-image-stub><span class="hidden lg:inline"></span>
        <button-stub class="!hidden md:!inline p-button-rounded p-button-sm p-button-text remove-additional-card-button" data-testid="recipient-role-additional-cards-multi-select-remove-pied-piper-for-thief"></button-stub></span>
        <!---->
      </div>
    </div>
    <div class="p-multiselect-dropdown" data-pc-section="dropdown">
      <chevron-down-icon-stub class="p-multiselect-dropdown-icon" aria-hidden="true" data-pc-section="dropdownicon"></chevron-down-icon-stub>
    </div>
    <portal-stub appendto="body" disabled="false"></portal-stub>
  </div>
</div>"
`;
